### PR TITLE
detach thread before erasing it from the FreeThreadPool

### PR DESCRIPTION
**Pull request by azat**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=ru

Category (leave one):
- Bug Fix

Short description (up to few sentences):

Otherwise std::thread will call std::terminate() from the dtor:

Detailed description (optional):

```
  ...
  #4  0x000000000ac829c1 in std::terminate() ()
  #5  0x0000000006f1c87f in std::thread::~thread (this=<optimized out>, __in_chrg=<optimized out>) at /usr/include/c++/7/thread:135
  ...
  #10 ThreadPoolImpl<std::thread>::worker (this=0xf370680 <ext::singleton<GlobalThreadPool>::instance()::instance>, thread_it=...) at ../dbms/src/Common/ThreadPool.cpp:191
```

Refs: f6b9b063071662f1c391ee95ff16a30c89069010 ("Attempt to implemnt global thread pool #4018")

### CODE REVIEW

File: dbms/src/Common/ThreadPool.cpp [hunk 1]
```diff
@@ -66,6 +66,7 @@ ReturnType ThreadPoolImpl<Thread>::scheduleImpl(Job job, int priority, std::opti
             }
             catch (...)
             {
+                threads.front().detach();
```
Comment (azat) [inline dbms/src/Common/ThreadPool.cpp]: Hm, this looks wrong, will rebase later

**Comment by alexey-milovidov** [comment]
The idea is correct but a fix is a little more complicated. Look at #4505.
